src/modules/users/user-types.ts

```diff
@@ -5,4 +5,8 @@ export interface UserName {
 
 export interface User extends UserName {
     id: number;
+}
+
+export interface AddUser {
+    (userName: UserName): void;
```

Comment:
Nemáš to špatně, ale běžnější je použít `type` pro typ funkce.

```suggestion
export type AddUser = (userName: UserName) => void;
```